src/engine.rs

```diff
@@ -30,22 +30,18 @@ fn import_modules(
 ) {
     let imported_modules: HashSet<String> =
         module.imports().map(|i| i.module().to_string()).collect();
-    imported_modules.iter().for_each(|imported_module| {
```

Comment:
Renamed `imported_module` to `module_name` because it was getting shadowed later.